src/main/java/uk/gov/hmcts/ccd/endpoint/ui/QueryEndpoint.java

```diff
@@ -337,4 +344,18 @@ private String validateCaseTypeId(String caseTypeId) {
         return caseTypeId;
     }
 
+    private boolean validateDate(List<String> datesToValidate) {
```

Comment:
1- Better mention all dates that are not valid in a BadRequestException, instead of just the first one.
2- Design-wise, let's double check this is inline with similar validations elsewhere in data store.
3- Let's check this will not be a breaking change for some date values currently treated as valid (perhaps checking prod logs, too).

Reply:
1 - Updated
2 - It is implemented in a similar way to the other endpoint validations in the data store
3 - Scanned through [Application Insights](https://portal.azure.com/#blade/AppInsightsExtension/BladeRedirect/BladeName/searchV1/ResourceId/%2Fsubscriptions%2F8999dec3-0104-4a27-94ee-6588559729d1%2FresourceGroups%2Fccd-shared-prod%2Fproviders%2Fmicrosoft.insights%2Fcomponents%2Fccd-prod/BladeInputs/%7B%22tables%22%3A%5B%22availabilityResults%22%2C%22requests%22%2C%22exceptions%22%2C%22pageViews%22%2C%22traces%22%2C%22customEvents%22%2C%22dependencies%22%5D%2C%22timeContextWhereClause%22%3A%22%7C%20where%20timestamp%20%3E%20datetime(%5C%222021-07-24T11%3A21%3A30.694Z%5C%22)%20and%20timestamp%20%3C%20datetime(%5C%222021-07-27T11%3A21%3A30.694Z%5C%22)%22%2C%22filterWhereClause%22%3A%22%7C%20where%20*%20has%20%5C%22GET%5C%22%20and%20*%20has%20%5C%22%2Faggregated%2Fcaseworkers%2F%3Auid%2Fjurisdictions%2FDIVORCE%2Fcase-types%2FDIVORCE%2Fcases%3Fview%3DSEARCH%26page%3D1%26%5C%22%7C%20order%20by%20timestamp%20desc%22%2C%22originalParams%22%3A%7B%22eventTypes%22%3A%5B%7B%22type%22%3A%22availabilityResult%22%2C%22tableName%22%3A%22availabilityResults%22%7D%2C%7B%22type%22%3A%22request%22%2C%22tableName%22%3A%22requests%22%7D%2C%7B%22type%22%3A%22exception%22%2C%22tableName%22%3A%22exceptions%22%7D%2C%7B%22type%22%3A%22pageView%22%2C%22tableName%22%3A%22pageViews%22%7D%2C%7B%22type%22%3A%22trace%22%2C%22tableName%22%3A%22traces%22%7D%2C%7B%22type%22%3A%22customEvent%22%2C%22tableName%22%3A%22customEvents%22%7D%2C%7B%22type%22%3A%22dependency%22%2C%22tableName%22%3A%22dependencies%22%7D%5D%2C%22timeContext%22%3A%7B%22durationMs%22%3A259200000%7D%2C%22filter%22%3A%5B%5D%2C%22searchPhrase%22%3A%7B%22originalPhrase%22%3A%22GET%20%2Faggregated%2Fcaseworkers%2F%3Auid%2Fjurisdictions%2FDIVORCE%2Fcase-types%2FDIVORCE%2Fcases%3Fview%3DSEARCH%26page%3D1%26%22%2C%22_tokens%22%3A%5B%7B%22conjunction%22%3A%22and%22%2C%22value%22%3A%22GET%22%2C%22isNot%22%3Afalse%2C%22kql%22%3A%22%20*%20has%20%5C%22GET%5C%22%22%7D%2C%7B%22conjunction%22%3A%22and%22%2C%22value%22%3A%22%2Faggregated%2Fcaseworkers%2F%3Auid%2Fjurisdictions%2FDIVORCE%2Fcase-types%2FDIVORCE%2Fcases%3Fview%3DSEARCH%26page%3D1%26%22%2C%22isNot%22%3Afalse%2C%22kql%22%3A%22%20and%20*%20has%20%5C%22%2Faggregated%2Fcaseworkers%2F%3Auid%2Fjurisdictions%2FDIVORCE%2Fcase-types%2FDIVORCE%2Fcases%3Fview%3DSEARCH%26page%3D1%26%5C%22%22%7D%5D%7D%2C%22sort%22%3A%22desc%22%7D%7D) and found only dates in the format acceptable to validation sent in the search are successful, all other date searches return 500's